core/src/main/java/com/xmartlabs/base/core/adapter/MillisecondsLocalDateAdapter.java

```diff
@@ -21,22 +23,36 @@
 
 /** {@link Gson} type adapter that serializes {@link LocalDate} objects to a millisecond format. */
 public class MillisecondsLocalDateAdapter implements JsonSerializer<LocalDate>, JsonDeserializer<LocalDate> {
+  @Nullable
   @Override
-  public synchronized JsonElement serialize(LocalDate date, Type type,
-                                            JsonSerializationContext jsonSerializationContext) {
+  public synchronized JsonElement serialize(@Nullable LocalDate date, @Nullable Type type,
+                                            @Nullable JsonSerializationContext jsonSerializationContext) {
     return Optional.ofNullable(date)
         .map(theDate -> new JsonPrimitive(theDate.atStartOfDay(ZoneOffset.UTC).toInstant().toEpochMilli()))
         .orElse(null);
   }
 
+  @Nullable
+  public synchronized JsonElement serialize(LocalDate date) {
```

Comment:
In my opinion it shouldn't be there. Another option is adding `@VisibleForTesting`